plugins/browser-plugin-accelerator-ecommerce/src/types.ts

```diff
@@ -0,0 +1,230 @@
+import { CommonEventProperties, SelfDescribingJson } from '@snowplow/tracker-core';
+
+/**
+ * Type/Schema for an ecommerce Action
+ */
+export interface Action {
+  /**
+   * Standard ecommerce actions
+   */
+  type:
+    | 'add_to_cart'
+    | 'remove_from_cart'
+    | 'product_view'
+    | 'list_click'
+    | 'list_view'
+    | 'checkout_step'
+    | 'transaction';
+
+  /**
+   * You can add a name for the list presented to the user.
+   * E.g. product list, search results, shop the look, frequently bought with
+   */
+  name?: string;
+}
+
+/**
+ * Type/Schema for a checkout_step event in Ecommerce
+ */
+export interface CheckoutStep {
+  /* Checkout step index */
+  step: number;
+  /* Shipping address postcode */
+  shipping_postcode?: string;
+  /* Billing address postcode */
+  billing_postcode?: string;
+  /* Full shipping address */
+  shipping_full_address?: string;
+  /* Full billing address */
+  billing_full_address?: string;
+  /* Can be used to discern delivery providers DHL, PostNL etc. */
+  delivery_provider?: string;
+  /* Store pickup, standard delivery, express delivery, international */
+  delivery_method?: string;
+  /* Coupon applied at checkout */
+  coupon_code?: string;
+  /* Selection of 'existing user' or 'guest checkout' */
+  account_type?: string;
+  /* Any kind of payment method the user selected to proceed. Card, PayPal, Alipay etc. */
+  payment_method?: string;
+  /* Invoice or receipt */
+  proof_of_payment?: string;
+  /* If opted in to marketing campaigns to the email address */
+  marketing_opt_in?: boolean;
+}
+
+/**
+ * Type for a cart entity in Ecommerce
+ */
+export interface Cart {
+  /**
+   * The unique ID representing this cart
+   */
+  cart_id?: string;
+  /**
+   * The total value of the cart after this interaction
+   */
+  total_value: number;
+  /**
+   * The currency used for this cart (ISO 4217)
+   */
+  currency: string;
+  /**
+   * Array of products added/removed from cart
```

Comment:
Now I see what confused me above, this explains that the `products` are only the changed products. Maybe naming this entity `CartChange` or something like that to invoke that it's the diff, not the actual "cart" would make it clearer for me.